src/fights.ts

```diff
@@ -2525,3 +2558,41 @@ function yachtzee(): void {
     }
   }
 }
+
+function runShadowRiftTurn(): void {
+  // we can probably have a better name
+  if (get("encountersUntilSRChoice") === 0) return;
+  if (globalOptions.prefs.yachtzeechain || get("rufusQuestType") === "items") {
```

Comment:
shouldn't it bypass the NC-forcer logic even if the rufus quest is the boss? given that garbo doesn't support the boss fights, i could see a small problem here where it would keep forcing a boss that garbo might not be able to kill if, for instance, the user came out of an ascension having accepted the boss fight quest. i doubt this came up in testing but i would just say != "artifact" instead of === items here